File: Python/DynamicProgramming/BJAlgo_1003.py
from sys import stdin

T = int(stdin.readline())

for _ in range(T):
    N = int(stdin.readline())

    cache_zero = [0 for index in range(N + 1)]
    cache_one = [0 for index in range(N + 1)]

    cache_zero[0] = 1
    cache_one[0] = 0

    if N >= 1:
        cache_zero[1] = 0
        cache_one[1] = 1

    if N > 1:
        for index in range(2, N + 1):
            cache_zero[index] = cache_zero[index - 1] + cache_zero[index - 2]
            cache_one[index] = cache_one[index - 1] + cache_one[index - 2]

    print("{} {}".format(cache_zero[N], cache_one[N]))

File: Java/DynamicProgramming/BJAlgo_1003.java
import java.io.*;

public class BJAlgo_1003 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int T = Integer.parseInt(br.readLine());

        for (int i = 0; i < T; i++) {
            int N = Integer.parseInt(br.readLine());
            int[] cache_zero = new int[N + 1];
            int[] cache_one = new int[N + 1];

            cache_zero[0] = 1;
            cache_one[0] = 0;

            if (N >= 1) {
                cache_zero[1] = 0;
                cache_one[1] = 1;
            }

            if (N > 1) {
                for (int j = 2; j < N + 1; j++) {
                    cache_zero[j] = cache_zero[j - 1] + cache_zero[j - 2];
                    cache_one[j] = cache_one[j - 1] + cache_one[j - 2];
                }
            }

            System.out.println(cache_zero[N] + " " + cache_one[N]);
        }
    }
}
